src/CacheClient.cpp

```diff
@@ -0,0 +1,198 @@
+#include "CacheClient.hpp"
+#include "Hash.hpp"
+#include "DagData.hpp"
+#include "RunAction.hpp"
+#include "NodeResultPrinting.hpp"
+#include "StatCache.hpp"
+#include "Profiler.hpp"
+#include "Exec.hpp"
+#include "BuildQueue.hpp"
+#include "MakeDirectories.hpp"
+
+const char* kENV_REAPI_CACHE_CLIENT = "REAPI_CACHE_CLIENT";
+const char* kENV_CACHE_SERVER_ADDRESS = "CACHE_SERVER_ADDRESS";
+const char* kENV_BEE_CACHE_BEHAVIOUR = "BEE_CACHE_BEHAVIOUR";
+
+static int SlowCallback(void *user_data, const char* label)
+{
+    SlowCallbackData *data = (SlowCallbackData *)user_data;
+    MutexLock(data->queue_lock);
+    char buffer[1000];
+    snprintf(buffer,sizeof(buffer),"%s %s", data->node_data->m_Annotation.Get(), label);
```

Comment:
Just checking: do we need to hold the lock when calling `data->node_data->m_Annotation.Get()` ?

Reply:
I believe no (the dag is a constant data structure in memory). But we need the lock when manipulating the build queue. However, it appears that `PrintNodeInProgress` ignores the build queue we pass in? So, probably we can simplify this, and remove the lock in this function (and from the `SlowCallbackData` struct) altogether. But that is code, which we did not add in this PR. @lucasmeijer , do you know about that?

Reply:
Actually, I think the lock here probably makes sense to keep for the sole purpose of making sure that log lines don't end up being printed in broken order or interleaved when two threads log output at the same time.

Reply:
The lock is surely needed for PrintNodeInProgress, but maybe not for the snprintf.